File: Python/BreakingRecords.py
def breakingRecords(scores):
    best, worst = 0, 0
    best_score, worst_score = scores[0], scores[0]
    length = len(scores)
    for i in range(1, length):
        if scores[i] > best_score:
            best_score = scores[i]
            best += 1
        if scores[i] < worst_score:
            worst_score = scores[i]
            worst += 1
    return [best, worst]

print(breakingRecords([3, 4, 21, 36, 10, 28, 35, 5, 24, 42]))
print(breakingRecords([10, 5, 20, 20, 4, 5, 2, 25, 1]))

File: JavaScript/breakingRecords.js
function breakingRecords(scores) {
    let best = 0;
    let worst = 0;
    let bestScore = scores[0];
    let worstScore = scores[0];
    const length = scores.length;
    for (let i = 1; i < length; i++) {
        if (scores[i] > bestScore) {
            bestScore = scores[i];
            best++;
        }
        if (scores[i] < worstScore) {
            worstScore = scores[i]
            worst++;
        }
    }
    return [best, worst];
}

console.log(breakingRecords([3, 4, 21, 36, 10, 28, 35, 5, 24, 42]))
console.log(breakingRecords([10, 5, 20, 20, 4, 5, 2, 25, 1]))